modules/testbase.lua

```diff
@@ -128,6 +128,29 @@ function control.runNextCase()
     xmlReporter.AddCase(Test.current_case_name)
     atf_logger.LOGTestCaseStart(Test.current_case_name)
     testcase(Test)
+
+    local function wait(pConnection)
+      local timeout = config.unexpectedEventTimeout
+      local event = events.Event()
+      event.matches = function(e1, e2) return e1 == e2 end
+
+      local ret = Expectation("Wait", pConnection)
+      ret.event = event
+      event_dispatcher:AddEvent(pConnection, event, ret)
+      Test:AddExpectation(ret)
+      ret:Timeout(timeout + 5000)
+      local function toRun()
+        event_dispatcher:RaiseEvent(pConnection, event)
+      end
+      RUN_AFTER(toRun, timeout)
```

Comment:
@dboltovskyi It is very risky to use global function which does not defined in current module in one of base modules in ATF. It creates implicit dependency on derived module `connecttest` and ATF architecture at all.
It also can lead to runtime errors which hard to investigate.

I would propose to define public function with only needed functionality in current module and extend it in derived module:
testbase module:
```
local function runAfter(self, func, timeout)
  local d = qt.dynamic()
  d.timeout = function(pTimer)
    func()
    self.timers[pTimer] = nil
  end
  
  local timer = timers.Timer()
  self.timers[timer] = true
  qt.connect(timer, "timeout()", d, "timeout()")
  timer:setSingleShot(true)
  timer:start(timeout)
end

...

local function main()
  setmetatable(Test, mt)

  qt.connect(control, "next()", control, "runNextCase()")

  rawset(Test, "FailTestCase", FailTestCase)
  rawset(Test, "SkipTest", SkipTest)
  rawset(Test, "RunAfter", runAfter)

  event_dispatcher = ed.EventDispatcher()

  ...

end
```
connecttest module:
```
function RUN_AFTER(func, timeout, funcName)
  local func_name_str = "noname"
  if funcName then
    func_name_str = funcName
  end
  xmlReporter.AddMessage(debug.getinfo(1, "n").name, func_name_str,
    {["functionLine"] = debug.getinfo(func, "S").linedefined, ["Timeout"] = tostring(timeout)})
  Test:RunAfter(func, timeout)
end
```

Reply:
@aderiabin Done in 0913260